### ISSUE TITLE
[SUGGESTION] Sprite:set_image for multiple textures

### ISSUE THREAD

**Issue by zendorx** [defold/defold#9037]
Problem:

I need to make look up shader for sprites. 
It will take color from tex0, calculate uv by this value and sample from another texture tex1 ("color scheme")
([example](
https://www.youtube.com/watch?v=HsOKwUwL1bE&list=PLZJ8e771dlAsCXytFC53e8AWHeWXD1d4P&index=3&ab_channel=aarthificial))

Problem apperas when I try to setup multiple textures for sprite. Tex1 ("color scheme") cannot contain sprites with same name as main texture (tex0).

There is a one workaround - pass texture in renderer using texture handler
```
local my_texture_info = resource.get_texture_info(my_texture_id)
..my_texture_info.handle
```
It will work only in case of one global tex1 ("color scheme"), but in case of need for multiple color schemes there is no solution. 

Suggestion: 
Extend API for [sprite](https://defold.com/ref/sprite/#image), add option to set image for specific texture by index
ex:
```
   go.set("#sprite", "atlas_image", {index = 0, image = "unit_orc"})
   go.set("#sprite", "atlas_image", {index = 1, image = "red_color_scheme"})
```

**Comment by AGulev** [comment]
Is it duplicate of https://github.com/defold/defold/issues/8609 ?

**Comment by zendorx** [comment]
> Is it duplicate of #8609 ?

Yes

**Comment by AGulev** [comment]
I've merged this issue with https://github.com/defold/defold/issues/8609 and closing this one

**Comment by AGulev** [comment]
fixed in 1.9.0 https://github.com/defold/defold/pull/9047
